### PR TITLE
print load and unload envs like direnv

### LINKED ISSUE #733: Show output when loading env vars
With direnv, when I'd `cd` into a dir with a `.envrc` file, I would get an output in my terminal showing which vars were loaded.

Similarly when I exited the folder there was an "Unloaded" message.

Generally I liked this/didn't consider it noise as it's good to know when your shell env vars are changing.

Having replaced direnv with rtx and switched to `.rtx.toml` to define the env vars, while the env vars get loaded/unloaded correctly I get no output in my terminal, even with `settings.verbose` enabled.

### PR DESCRIPTION
**Pull request by dfang**
close #733

@jdxcode  maybe add an ENV to toggle this, like `RTX_ENV_LOAD_MESSAGE` or else?

### CODE REVIEW

File: src/hook_env.rs [hunk 1]
```diff
@@ -158,9 +158,15 @@ pub fn build_env_commands(shell: &dyn Shell, patches: &EnvDiffPatches) -> String
     for patch in patches.iter() {
         match patch {
             EnvDiffOperation::Add(k, v) | EnvDiffOperation::Change(k, v) => {
+                if !k.starts_with("__RTX") && k != "PATH" {
+                    info!("load env {}={}", k, v);
```
Comment (jdx) [inline src/hook_env.rs]: This is not the right place for this code

**Review by jdxcode — CHANGES_REQUESTED** [review]

File: src/hook_env.rs (hunk 1 above)
Comment (jdx) [reply src/hook_env.rs]: this function is just a utility function to build the change list, it shouldn't be displaying any UI, that should be in the command

File: src/cli/hook_env.rs [hunk 2]
```diff
@@ -54,6 +54,21 @@ impl Command for HookEnv {
         patches.push(self.build_diff_operation(&diff)?);
         patches.push(self.build_watch_operation(&config)?);
 
+        // output load and unload env like direnv
+        for patch in patches.iter() {
```
Comment (jdx) [inline src/cli/hook_env.rs:75]: This should go inside of display_status() I think

File: src/cli/hook_env.rs [hunk 3]
```diff
@@ -65,7 +55,47 @@ impl Command for HookEnv {
...
+                EnvDiffOperation::Remove(k) if !k.starts_with("__RTX") && k != "PATH" => {
+                    info!("unload env {}", k);
+                }
+                _ => {}
+            }
+        }
+    }
+
     fn display_status(&self, config: &Config, ts: &Toolset, out: &mut Output) {
+        self.display_env_load_unload_status(config, ts);
```
Comment (jdx) [inline src/cli/hook_env.rs]: can we pass `patches` through so we don't need to regenerate it?

Comment (jdx) [reply src/cli/hook_env.rs]: keep the extraction that added `get_patches()` though, that's a useful refactor

File: src/cli/hook_env.rs [hunk 4]
```diff
@@ -65,7 +55,47 @@ impl Command for HookEnv {
...
+        patches.push(self.build_watch_operation(config)?);
+
+        Ok(patches)
+    }
+
+    fn display_env_load_unload_status(&self, config: &Config, ts: &Toolset) {
+        let patches = match self.get_patches(config, ts) {
+            Ok(patches) => patches,
+            Err(error) => {
+                eprintln!("Error: {}", error);
```
Comment (jdx) [inline src/cli/hook_env.rs]: don't ever say "error" unless something actually failed, this should say "warn" and use `warn!()`

File: src/cli/hook_env.rs [hunk 5]
```diff
@@ -65,7 +55,47 @@ impl Command for HookEnv {
...
+                return;
+            }
+        };
+
+        for patch in patches {
+            match patch {
+                EnvDiffOperation::Add(k, v) | EnvDiffOperation::Change(k, v) if !k.starts_with("__RTX") && k != "PATH" => {
+                    info!("load env {}={}", k, v);
```
Comment (jdx) [inline src/cli/hook_env.rs]: don't display the value, the value could be massive. I think we should use direnv's syntax here for `+ADDED` `-REMOVED` `~CHANGED` https://github.com/direnv/direnv/blob/517e9f5b4159db2ce63695e4f51ca7b38b867876/internal/cmd/cmd_export.go#L102

Comment (jdx) [reply src/cli/hook_env.rs]: also show everything on one line

**Pull request closed without merging**